### DISCUSSION TITLE
plans for BGP as build in feature

### DISCUSSION THREAD

**Discussion by AleksBelytskyi** [netbox-community/netbox#14826] [Q&A] [answered]
Hello folks,

Were there any ideas or discussions to implement BGP features as the built-in feature of Netbox?

I know there is a plugin that extends Netbox to support BGP related things, but one has to be sure that it will or will not be implemented directly into the core functionality of Netbox, as happened with the [Tunnels plugin](https://github.com/robertlynch3/netbox-tunnels2) recently. 

Thanks in advance.

**Comment by jeremystretch** [comment, marked as answer]
There aren't currently any plans to incorporate BGP modeling in NetBox, but you're of course welcome to open a [feature request](https://github.com/netbox-community/netbox/issues/new?assignees=&labels=type%3A+feature&projects=&template=feature_request.yaml) for it.

> but one has to be sure that it will or will not be implemented directly into the core functionality of Netbox, as happened with the [Tunnels plugin](https://github.com/robertlynch3/netbox-tunnels2) recently.

FYI the work implemented under #9816 to support VPN tunnel modeling in NetBox v3.7 was entirely original. Even the underlying data model differs greatly from the plugin. I point this out to ensure anyone currently using the plugin does not assume that they can migrate data from that plugin directly to NetBox v3.7.

**Reply by AleksBelytskyi** [reply]
Hi @jeremystretch 

Thank you for the prompt reply.

> I point this out to ensure anyone currently using the plugin does not assume that they can migrate data from that plugin directly to NetBox v3.7.

That's why I'm asking since we're currently deciding whether we should move to the built-in feature or continue using the Tunnels plugin.
Personally I for using native features but there is a lot of done around our current schema so a lot of must be done to migrate.

We also want to move the BGP configuration to Netbox and avoid "rebranding" in the next few years.

Anyway, thank you very much for the reply @jeremystretch Your answer completely satisfied me. Thank you for such a great project that you do for all of us.

Have a nice day.
